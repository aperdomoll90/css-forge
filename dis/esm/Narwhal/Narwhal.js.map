{"version":3,"file":"Narwhal.js","sourceRoot":"","sources":["../../../src/Narwhal/Narwhal.tsx"],"names":[],"mappings":";;;;;;;;;;;;AAAA,OAAO,cAAc,CAAA;AACrB,OAAO,2BAA2B,CAAA;AAClC,OAAO,WAAW,MAAM,0BAA0B,CAAA;AAClD,OAAO,OAAO,MAAM,sBAAsB,CAAA;AAC1C,OAAO,GAAG,MAAM,kBAAkB,CAAA;AAElC,MAAM,CAAC,IAAM,OAAO,GAAiB;IACnC,OAAO,CACL,wBAAK,SAAS,EAAC,aAAa,iBAC1B,uBAAK,SAAS,EAAC,OAAO,gBACpB,uCAAmB,IACf,EACN,wBAAK,SAAS,EAAC,aAAa,iBAC1B,cAAK,GAAG,EAAE,OAAO,EAAE,SAAS,EAAC,cAAc,EAAC,GAAG,EAAC,SAAS,GAAG,EAC5D,cAAK,GAAG,EAAE,GAAG,EAAE,SAAS,EAAC,KAAK,EAAC,GAAG,EAAC,KAAK,GAAG,EAC3C,cAAK,GAAG,EAAE,WAAW,EAAE,SAAS,EAAC,aAAa,EAAC,GAAG,EAAC,cAAc,GAAG,EACpE,cAAK,GAAG,EAAE,OAAO,EAAE,SAAS,EAAC,SAAS,EAAC,GAAG,EAAC,SAAS,GAAG,EACvD,wBAAK,SAAS,EAAC,UAAU,iBACvB,cAAK,SAAS,EAAC,eAAe,GAAG,EACjC,cAAK,SAAS,EAAC,eAAe,GAAG,KAC7B,KACF,EACN,cAAK,SAAS,EAAC,eAAe,GAAG,KAC7B,CACP,CAAA;AACH,CAAC,CAAA"}