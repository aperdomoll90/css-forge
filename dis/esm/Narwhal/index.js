export { Narwhal } from "./Narwhal";
//# sourceMappingURL=index.js.map