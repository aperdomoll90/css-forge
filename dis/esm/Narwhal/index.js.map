{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/Narwhal/index.tsx"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,WAAW,CAAA"}