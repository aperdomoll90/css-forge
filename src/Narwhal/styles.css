.narwhalArea {
  position: relative;
  width: 220px;
  height: 220px;
  border-radius: 10px;
  background-color: rgba(108, 108, 248, 0.445);
  box-shadow: 2px 2px 10px rgba(0, 0, 0, 0.466);
}

.shade{
    opacity: 0;
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    z-index: 10;
    background-color: rgba(0, 0, 0, 0.733);
    display: flex;
    align-items: center;
    justify-content: center;
    color: white;
    font-size: 2rem;
    font-weight: 900;
    letter-spacing: 5px;
    text-transform: uppercase;
}
.shade:hover{
    opacity: 1;
}

/* .mainBody{
    width: 200px;
    height: 200px;
    position: absolute;
    top:calc(50% - 100px); 
    left:calc(50% - 100px); 
    border-radius: 50%;
    background-color: rgb(163, 163, 245);
} */

.fullNarwhal {
  z-index: 2;
  width: 200px;
  height: 200px;
  position: relative;
  top: 5%;
  animation: floating 2s infinite ease-in-out alternate;
}
@keyframes floating {
  0% {
    transform: translateY(10%);
  }
  50% {
    transform: translateY(-15%);
  }
}

.narwhalBody {
  z-index: 5;
  width: 200px;
  position: absolute;
  right: 0;
}

.flipper {
  z-index: 7;
  width: 35px;
  position: absolute;
  top: 90px;
  right: 80px;
  animation: leftFlipping 2s infinite ease-in-out alternate;
}

@keyframes leftFlipping {
  0% {
    transform: rotate(1deg);
  }
  50% {
    transform: rotate(21deg);
  }
}

.flipperRight {
  width: 40px;
  position: absolute;
  top: 90px;
  right: 0px;
  transform: scaleX(-1) rotate(11deg);
  z-index: 0;
  animation: rightFlipping 2s infinite ease-in-out alternate;
}
@keyframes rightFlipping {
  0% {
    transform: scaleX(-1) rotate(-21deg);
  }
  50% {
    transform: scaleX(-1) rotate(-1deg);
  }
}

.fin {
  width: 83px;
  position: absolute;
  left: 2px;
  top: 36px;
  z-index: 4;
  transform: rotate(9deg);
  animation: tailFlipping 2s infinite ease-in-out alternate;
}

@keyframes tailFlipping {
  /* 0% {
    transform: skewY(10deg);
  } */
  50% {
    transform: skewY(20deg);
  }
}

.narwhalShadow {
  z-index: 0;
  width: 110px;
  height: 20px;
  position: absolute;
  bottom: 20px;
  left: 60px;
  border-radius: 50%;
  background-color: rgba(0, 0, 0, 0.671);
  animation: narwhalShadow 2s infinite ease-in-out alternate;
}
@keyframes narwhalShadow {
  0% {
    transform: scale(100%);
  }
  50% {
    transform: scale(60%);
    opacity: 60%;
  }
}

.blinkers {
  z-index: 9;
  width: 52px;
  position: absolute;
  right: 20px;
  top: 74.5px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  animation: blinking 2s infinite ease-in-out alternate;
}

.narwhalCircle {
  width: 11.5px;
  height: 12px;
  border-radius: 50%;
  background-color: #b6eaff;
}

@keyframes blinking {
  0% {
    transform: translateY(100%);
  }
  50% {
    transform: translateY(0%);
  }
}
