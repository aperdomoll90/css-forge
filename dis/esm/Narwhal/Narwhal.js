var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import './styles.css';
import '../utils/GlobalStyles.css';
import narwhalBody from './images/narwhalBody.png';
import flipper from './images/flipper.png';
import fin from './images/fin.png';
export var Narwhal = function () {
    return (_jsxs("div", __assign({ className: 'narwhalArea' }, { children: [_jsx("div", __assign({ className: 'shade' }, { children: _jsx("p", { children: "Nora Narwhal" }) })), _jsxs("div", __assign({ className: 'fullNarwhal' }, { children: [_jsx("img", { src: flipper, className: 'flipperRight', alt: 'flipper' }), _jsx("img", { src: fin, className: 'fin', alt: 'fin' }), _jsx("img", { src: narwhalBody, className: 'narwhalBody', alt: 'Narwhal Body' }), _jsx("img", { src: flipper, className: 'flipper', alt: 'flipper' }), _jsxs("div", __assign({ className: 'blinkers' }, { children: [_jsx("div", { className: 'narwhalCircle' }), _jsx("div", { className: 'narwhalCircle' })] }))] })), _jsx("div", { className: 'narwhalShadow' })] })));
};
//# sourceMappingURL=Narwhal.js.map