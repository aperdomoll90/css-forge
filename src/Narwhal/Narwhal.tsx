import './styles.css'
import '../utils/GlobalStyles.css'
import narwhalBody from './images/narwhalBody.png'
import flipper from './images/flipper.png'
import fin from './images/fin.png'

export const Narwhal: React.FC<{}> = () => {
  return (
    <div className='narwhalArea'>
      <div className='shade'>
        <p>Nora Narwhal</p>
      </div>
      <div className='fullNarwhal'>
        <img src={flipper} className='flipperRight' alt='flipper' />
        <img src={fin} className='fin' alt='fin' />
        <img src={narwhalBody} className='narwhalBody' alt='Narwhal Body' />
        <img src={flipper} className='flipper' alt='flipper' />
        <div className='blinkers'>
          <div className='narwhalCircle' />
          <div className='narwhalCircle' />
        </div>
      </div>
      <div className='narwhalShadow' />
    </div>
  )
}
