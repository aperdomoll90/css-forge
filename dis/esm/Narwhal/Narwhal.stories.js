import { jsx as _jsx } from "react/jsx-runtime";
import { Narwhal } from './Narwhal';
import StorybookContainer from '../StorybookContainer';
export default {
    title: 'Petting Zoo',
    component: Narwhal,
    parameters: {
        jest: ['Narwhal.test.tsx'],
    },
};
var NarwhalTemplate = function () { return (_jsx(StorybookContainer, { children: _jsx(Narwhal, {}) })); };
export var NarwhalComponent = NarwhalTemplate.bind({});
//# sourceMappingURL=Narwhal.stories.js.map