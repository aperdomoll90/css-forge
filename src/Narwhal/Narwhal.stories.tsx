import { Meta, StoryFn } from '@storybook/react'
import {Narwhal} from './Narwhal'
import StorybookContainer from '../StorybookContainer'

export default {
  title: 'Petting Zoo',
  component: Narwhal,
  parameters: {
    jest: ['Narwhal.test.tsx'],
  },
} as Meta

const NarwhalTemplate: StoryFn = () => (
  <StorybookContainer>
    <Narwhal />
  </StorybookContainer>
)

export const NarwhalComponent = NarwhalTemplate.bind({})
