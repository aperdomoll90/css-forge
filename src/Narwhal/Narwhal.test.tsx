import { render } from '@testing-library/react'
import {Narwhal} from './Narwhal'

describe('Narwhal component', () => {
  // const { axe, toHaveNoViolations } = require('jest-axe')
  // expect.extend(toHaveNoViolations)

  const renderComponent = () => render(<Narwhal />)
  it('should match snapshots', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
  //   it('should demonstrate ADA compliance on this component', async () => {
  //     const { container } = renderComponent()
  //     const results = await axe(container)
  //     expect(container).toHaveNoViolations()
  //   })
})
