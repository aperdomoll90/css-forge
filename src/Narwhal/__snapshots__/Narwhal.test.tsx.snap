// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Narwhal component should match snapshots 1`] = `
<div>
  <div
    class="narwhalArea"
  >
    <div
      class="shade"
    >
      <p>
        Nora Narwhal
      </p>
    </div>
    <div
      class="fullNarwhal"
    >
      <img
        alt="flipper"
        class="flipperRight"
        src="flipper.png"
      />
      <img
        alt="fin"
        class="fin"
        src="fin.png"
      />
      <img
        alt="Narwhal Body"
        class="narwhalBody"
        src="narwhalBody.png"
      />
      <img
        alt="flipper"
        class="flipper"
        src="flipper.png"
      />
      <div
        class="blinkers"
      >
        <div
          class="narwhalCircle"
        />
        <div
          class="narwhalCircle"
        />
      </div>
    </div>
    <div
      class="narwhalShadow"
    />
  </div>
</div>
`;
